### PR TITLE
CUDA BFloat16 signal windows

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Looks like this op is never tested for the support of different dtypes?

### CODE REVIEW

**Comment by ngimel** [comment]
it is tested in `test_signal_window_functions`, please update the test accordingly.

**Comment by zasdfgbnm** [comment]
test added for all dtypes, and arange is required for hann. arange will be tested in https://github.com/pytorch/pytorch/pull/44848

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu [hunk 1]
```diff
@@ -225,11 +225,9 @@ void nan_to_num_kernel_cuda(
...
 void kaiser_window_kernel_cuda(TensorIterator& iter, int64_t window_length, double beta){
   AT_DISPATCH_FLOATING_TYPES_AND2(ScalarType::Half, ScalarType::BFloat16, iter.dtype(), "kaiser_window_cuda", [&](){
-    AT_SKIP_BFLOAT16_IF_NOT_ROCM(scalar_t, "kaiser_window_cuda", [&] {
-      const scalar_t alpha = static_cast<scalar_t>((window_length - 1) / 2.0);
-      gpu_kernel(iter, [=]GPU_LAMBDA(scalar_t a) -> scalar_t {
-        return calc_i0(static_cast<scalar_t>(beta) * ::sqrt(1 - ::pow((a - alpha) / alpha, static_cast<scalar_t>(2.0)))) / calc_i0(static_cast<scalar_t>(beta));
-      });
+    const scalar_t alpha = static_cast<scalar_t>((window_length - 1) / 2.0);
+    gpu_kernel(iter, [=]GPU_LAMBDA(scalar_t a) -> scalar_t {
+      return calc_i0(static_cast<scalar_t>(beta) * ::sqrt(1 - ::pow((a - alpha) / alpha, static_cast<scalar_t>(2.0)))) / calc_i0(static_cast<scalar_t>(beta));
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/UnaryOpsKernel.cu:230]: the intermediate computations of i0 args here should still be in accscalar_t? I can merge as though.

**Review by ngimel — APPROVED** [review]

File: aten/src/ATen/native/cuda/UnaryOpsKernel.cu (hunk 1 above)
Comment (zasdfgbnm) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu:230]: Is i0 bw bound or compute bound? If it is not bw bound, does it still make sense to compute on accscalar_t?

Comment (ngimel) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu:230]: I don't know, but it's doing internal computations in fp32 anyway, and here 
```
static_cast<scalar_t>(beta) * ::sqrt(1 - ::pow((a - alpha) / alpha
```
is still doing repeated conversions and truncations

Comment (zasdfgbnm) [reply aten/src/ATen/native/cuda/UnaryOpsKernel.cu:230]: https://github.com/pytorch/pytorch/blob/719d29dab5e915437f026a01feb45fef62828047/aten/src/ATen/native/Math.h#L542
OK, `calc_i0` is already computing on float32 anyway

**Comment by zasdfgbnm** [comment]
ping @ngimel

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**